File: problems/1390_four_divisors/1390_four_divisors.py
class Solution:
    def sumFourDivisors(self, nums: List[int]) -> int:
        total_sum = 0
        for num in nums:
            cur_sum = 0
            count = 0

            for i in range(1, int(num ** 0.5) + 1):
                if num % i == 0:
                    if i * i == num: 
                        count += 1
                        cur_sum += i
                    else:
                        count += 2
                        cur_sum += (i + num // i)
                if count > 4:
                    break
            if count == 4:
                total_sum += cur_sum
        return total_sum

File: problems/1390_four_divisors/1390_four_divisors.java
class Solution {
    public int sumFourDivisors(int[] nums) {
        int totalSum = 0;
        for (int num : nums) {
            int curSum = 0;
            int count = 0;
            for (int i = 1; i * i <= num; i++) {
                if (num % i == 0) {
                    if (i * i == num) {
                        count++;
                        curSum += i;
                    } else {
                        count += 2;
                        curSum += (i + num / i);
                    }
                }

                if (count > 4)
                    break;
            }
            if (count == 4)
                totalSum += curSum;
        }
        return totalSum;
    }
}